spec/index.bs

```diff
@@ -840,6 +881,66 @@ enum GPUBindingType {
+        1. Set |layoutBinding|.{{layout binding/[[textureComponentType]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/textureComponentType}}.
+        1. Set |layoutBinding|.{{layout binding/[[multisampled]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/multisampled}}.
+    1. Set |layoutBinding|.{{layout binding/[[binding]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/binding}}.
+    1. Set |layoutBinding|.{{layout binding/[[visibility]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/visibility}}.
+    1. Set |layoutBinding|.{{layout binding/[[type]]}} to the value of |bindingDescriptor|.{{GPUBindGroupLayoutBinding/type}}.
+    1. Insert |layoutBinding| into |layout|.{{GPUBindGroupLayout/[[bindingsMap]]}} with a key of |layoutBinding|.{{layout binding/[[binding]]}}.
+1. Return |layout|.
+
+<b>Validation</b>
+* The device must not be lost.
+* Each |bindingDescriptor|.{{GPUBindGroupLayoutBinding/binding}} in |descriptor| must be unique.
+* There must be {{GPULimits/maxUniformBuffersPerShaderStage|GPULimits.maxUniformBuffersPerShaderStage}} or 
+    fewer |bindingDescriptor|s of type {{GPUBindingType/uniform-buffer}} visible on each shader stage in |descriptor|.
+    * There must be {{GPULimits/maxDynamicUniformBuffersPerPipelineLayout|GPULimits.maxDynamicUniformBuffersPerPipelineLayout}} or
+        fewer |bindingDescriptor|s of type {{GPUBindingType/uniform-buffer}} with {{GPUBindGroupLayoutBinding/hasDynamicOffsets}} set to `true` 
+        visible across <b>all</b> shader stages in |descriptor|.
+* There must be {{GPULimits/maxStorageBuffersPerPipelineLayout|GPULimits.maxStorageBuffersPerPipelineLayout}} or 
+    fewer |bindingDescriptor|s of type {{GPUBindingType/storage-buffer}} visible across <b>all</b> shader stages in |descriptor|.
+    * There must be {{GPULimits/maxDynamicStorageBuffersPerPipelineLayout|GPULimits.maxDynamicStorageBuffersPerPipelineLayout}} or
+        fewer |bindingDescriptor|s of type {{GPUBindingType/storage-buffer}} with {{GPUBindGroupLayoutBinding/hasDynamicOffsets}} set to `true` 
+        visible across <b>all</b> shader stages in |descriptor|.
+* There must be {{GPULimits/maxSampledTexturesPerShaderStage|GPULimits.maxSampledTexturesPerShaderStage}} or 
+    fewer |bindingDescriptor|s of type {{GPUBindingType/sampled-texture}} visible on each shader stage in |descriptor|.
+* There must be {{GPULimits/maxStorageTexturesPerShaderStage|GPULimits.maxStorageTexturesPerShaderStage}} or 
+    fewer |bindingDescriptor|s of type {{GPUBindingType/storage-texture}} visible on each shader stage in |descriptor|.
+* There must be {{GPULimits/maxSamplersPerShaderStage|GPULimits.maxSamplersPerShaderStage}}  or 
+    fewer |bindingDescriptor|s of type {{GPUBindingType/sampler}} visible on each shader stage in |descriptor|.
+
+At any point in the algorithm, if any of the above conditions are violated, run the following steps:
```

Comment:
Instead of "at any point", it makes more sense to me to specify when these checks are done. (Otherwise a naive implementation supposedly has to run all of these checks after every step.)

If there's a step that checks if these conditions are validated, these two steps could be inlined under that.

Reply:
I'm having trouble deciding the best way to reference the very long validation conditions in-line. I'd like to leave them where they are but point to them (e.g. "If [condition 1] is violated, return with error"). How should I link "condition 1" to its definition?

Reply:
Let's worry about this a little later when we have some more experience writing these rules and then refactor them to be consistent. I think it's clear enough for right now.